src/streaming/text/TextController.js

```diff
@@ -214,6 +221,19 @@ function TextController() {
         textTracksAdded = true;
     }
 
+    function onCurrentTrackChanged(event) {
+        if (!initialSettingsSet && event && event.newMediaInfo) {
```

Comment:
Does it make sense to override the default settings even if initialSettingsSet = true? Track Change should be caused by an explicit change of the default track (user interaction).

Reply:
In fact IMO we shouldn't update default settings when user is changing track.
If a default lang is set by application, this is up to the application to decide wether or not the default lang must be updated if the user is changing the track during playback.